### PR TITLE
feat:Added Implementation of Split_Largest_Array_Sum Algorithm

### PR DESCRIPTION
**Pull request by ARYPROGRAMMER**
#### Description of Change

Added the Solution to LeetCode (Medium-DP) Question "Split Largest Sum Array"

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Uses O(n) efficient approach to handle all test cases

### CODE REVIEW

File: dynamic_programming/split_array_largest_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    * Explanation: There are four ways to split nums into two subarrays.
+    * The best way is to split it into [1,2,3] and [4,5], where the largest sum
+    * among the two subarrays is only 9.
+
+ * @author [ARYA PRATAP SINGH](https://github.com/ARYPROGRAMMER)
+ * [Leetcode](https://leetcode.com/problems/split-array-largest-sum/description/?envType=problem-list-v2&envId=dynamic-programming)
+ */
+
+// header files
+#include <iostream>
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:29]: document headers using https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md

File: dynamic_programming/split_array_largest_sum.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    namespace split_array_largest_sum{
+
+        /**
+         * @brief This function calculates the minimum largest sum of the split
+         * @param nums integer array
+         * @param k integer
+         * @
+         */
+
+        int dp[1003][53];
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:51]: use array from stl

File: dynamic_programming/split_array_largest_sum.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    };
+    // custom value of k
+    int k = 5;
+   
+    //calling the function
+    int ans = dynamic_programming::split_array_largest_sum::splitArray(v, k);
+
+    // expected output
+    int expectedOutput = 30;
+
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:97]: redundant variables

File: dynamic_programming/split_array_largest_sum.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+    int k;
+    std::cout << "Enter the value of k : ";
+    std::cin >> k;
+
+    int ans;
+
+    // user output
+    ans = dynamic_programming::split_array_largest_sum::splitArray(v, k);
+    std::cout << "The minimum largest sum of the split is : " << ans << std::endl;
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:129]:
```suggestion
```
no code inside main function please

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by ARYPROGRAMMER — COMMENTED** [review]
tried to solve all changes, waiting for re-review

File: dynamic_programming/split_array_largest_sum.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+#include <vector>
+#include <climits>
+#include <cstring>
+#include <cassert>
+
+namespace dynamic_programming {
+    namespace split_array_largest_sum {
+
+        // DP table for memoization
+        int dp[1003][53];
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:47]: Should use std::array

File: dynamic_programming/split_array_largest_sum.cpp [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+#include <iostream>
+#include <vector>
+#include <climits>
+#include <cstring>
+#include <cassert>
+
+namespace dynamic_programming {
+    namespace split_array_largest_sum {
+
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:45]: File should be linted using clang tidy/ format

File: dynamic_programming/split_array_largest_sum.cpp [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+ * [ARYA PRATAP SINGH](https://github.com/ARYPROGRAMMER)
+ */
+
+#include <iostream>
+#include <vector>
+#include <climits>
+#include <cstring>
+#include <cassert>
+
+namespace dynamic_programming {
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:43]: Namespaces should be documented

File: dynamic_programming/split_array_largest_sum.cpp [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+ * O(n^2 * k), where `n` is the size of the array and `k` is the number of subarrays.
+ * 
+ * ### Space Complexity:
+ * O(n * k), for the DP table.
+ * 
+ * @author 
+ * [ARYA PRATAP SINGH](https://github.com/ARYPROGRAMMER)
+ */
+
+#include <iostream>
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:37]: Headers should be documented

File: dynamic_programming/split_array_largest_sum.cpp [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+/**
+ * @brief Main Function (driver code)
+ * This function calls test cases and handles user input/output.
+ */
+int main() {
+    // Running test cases
+    test();
+
+    // Handling user input/output
+    handleUserIO();
```
Comment (realstealthninja) [inline dynamic_programming/split_array_largest_sum.cpp:133]: No user inputs please, would make it untestable

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by ARYPROGRAMMER — COMMENTED** [review]
@realstealthninja please review and verify changes

**Pull request closed without merging**